packages/core/addon/chart-builders/dimension.ts

```diff
@@ -0,0 +1,166 @@
+import { computed } from '@ember/object';
+import { assert } from '@ember/debug';
+import moment, { MomentInput } from 'moment';
+import DataGroup from 'navi-core/utils/classes/data-group';
+import { API_DATE_FORMAT_STRING } from 'navi-data/utils/date';
+//@ts-ignore
+import tooltipLayout from '../templates/chart-tooltips/dimension';
+import ChartAxisDateTimeFormats from 'navi-core/utils/chart-axis-date-time-formats';
+import { getRequestDimensions, groupDataByDimensions } from 'navi-core/utils/chart-data';
+import BaseChartBuilder from './base';
+import RequestFragment from 'navi-core/models/bard-request-v2/request';
+import { ResponseV1 } from 'navi-data/serializers/facts/interface';
+import { tracked } from '@glimmer/tracking';
+import { DimensionSeriesConfig } from '../models/chart-visualization';
+import ColumnFragment from '../models/bard-request-v2/fragments/column';
+
+type ResponseRow = ResponseV1['rows'][number];
+
+export default class DimensionChartBuilder extends BaseChartBuilder {
+  @tracked byXSeries?: DataGroup<ResponseRow>;
+
+  /**
+   * @param row - single row of fact data
+   * @param config - series config
+   * @param request - request used to query fact data
+   * @returns name of series given row belongs to
+   */
+  getSeriesName(row: ResponseRow, _config: unknown, request: RequestFragment): string {
+    // let dimensionOrder = config.dimensionOrder;
+    return request.dimensionColumns.map(dim => row[dim.canonicalName]).join(',');
+  }
+
+  /**
+   * @inheritdoc
+   */
+  getXValue(row: ResponseRow, _config: unknown, request: RequestFragment): string {
+    // expects timeGrainColumn values to be a readable moment input
+    const date = row[request.timeGrainColumn.canonicalName] as MomentInput;
+    return moment(date).format(API_DATE_FORMAT_STRING);
+  }
+
+  /**
+   * @inheritdoc
+   */
+  buildData(response: ResponseV1, config: DimensionSeriesConfig, request: RequestFragment) {
+    const timeGrainColumn = request.timeGrainColumn.canonicalName;
+    const { timeGrain, interval } = request;
+    assert('request should have an interval', interval);
+    assert('request should have a timeGrain', timeGrain);
+    // Group data by x axis value + series name in order to lookup trends when building tooltip
+    this.byXSeries = new DataGroup(response.rows, row => {
+      return `${this.getXValue(row, config, request)} ${this.getSeriesName(row, config, request)}`;
+    });
+
+    // Support different `dateTime` formats by mapping them to a standard
+    const buildDateKey = (dateTime: MomentInput) =>
+      timeGrain !== undefined && timeGrain !== 'all'
+        ? moment(dateTime)
+            .startOf(timeGrain)
+            .format(API_DATE_FORMAT_STRING)
+        : moment(dateTime).format(API_DATE_FORMAT_STRING);
+
+    const { metric: metricCid } = config;
+    const metric = request.columns.find(c => c.cid === metricCid) as ColumnFragment;
+    const dimensions = getRequestDimensions(request);
```

Comment:
It's kind of a weird Fili-ism that's stuck around because we have a `dimensionColumns` property on the request fragment already, but `getRequestDimensions` filters out `dateTime` explicitly.

Reply:
#1074